src/sage/homology/homology_vector_space_with_basis.py

```diff
@@ -676,6 +937,96 @@ def cup_product(self, other):
             """
             return self * other
 
+
+class CohomologyRing_mod2(CohomologyRing):
+    r"""
+    The mod 2 cohomology ring.
+
+    Based on :class:`CohomologyRing`, with Steenrod operations included.
+
+    .. NOTE::
+
+        This is not intended to be created directly by the user, but
+        instead via the
+        :meth:`cohomology ring<sage.topology.cell_complex.GenericCellComplex.cohomology_ring>`
+        of a :class:`cell
+        complex<sage.topology.cell_complex.GenericCellComplex>`.
+
+    .. TODO::
+
+        Implement Steenrod operations on (co)homology at odd primes,
+        and thereby implement this class over `\GF{p}` for any `p`.
+
+    INPUT:
+
+    - ``base_ring`` -- must be the field ``GF(2)``
+    - ``cell_complex`` -- the cell complex whose homology we are
+      computing
+    - ``category`` -- (optional) a subcategory of modules with basis
+
+    EXAMPLES:
+
+    Mod 2 cohomology operations are defined on both the left and the
+    right::
+
+        sage: CP2 = simplicial_complexes.ComplexProjectivePlane()
+        sage: Hmod2 = CP2.cohomology_ring(GF(2))
+        sage: y = Hmod2.basis(2)[2,0]
+        sage: y.Sq(2)
+        h^{4,0}
+
+        sage: # needs sage.groups
+        sage: Y = simplicial_sets.RealProjectiveSpace(6).suspension()
+        sage: H_Y = Y.cohomology_ring(GF(2))
+        sage: b = H_Y.basis()[2,0]
+        sage: b.Sq(1)
+        h^{3,0}
+        sage: b.Sq(2)
+        0
+        sage: c = H_Y.basis()[4,0]
+        sage: c.Sq(1)
+        h^{5,0}
+        sage: c.Sq(2)
+        h^{6,0}
+        sage: c.Sq(3)
+        h^{7,0}
+        sage: c.Sq(4)
+        0
+
+    Cohomology can be viewed as a left module over the Steenrod
+    algebra, and also as a right module::
+
+        sage: # needs sage.groups
+        sage: RP4 = simplicial_sets.RealProjectiveSpace(4)
+        sage: H = RP4.cohomology_ring(GF(2))
+        sage: x = H.basis()[1,0]
+        sage: Sq(0,1) * x
+        h^{4,0}
+        sage: Sq(3) * x
+        0
+        sage: x * Sq(3)
+        h^{4,0}
+    """
+
+    def __init__(self, base_ring, cell_complex):
+        """
+        Initialize ``self``.
+
+        EXAMPLES::
+
+            sage: RP2 = simplicial_complexes.ProjectivePlane()
+            sage: H = RP2.cohomology_ring(GF(2))
+            sage: TestSuite(H).run()
+        """
+        if base_ring != GF(2):
+            raise ValueError
```

Comment:
```suggestion
            raise ValueError("the base ring must be GF(2)")
```

Reply:
This should never happen — the class should only ever be instantiated when the base ring is GF(2) — but the change is fine.